observations/obs_converters/even_sphere/even_sphere_subdom.m

```diff
@@ -0,0 +1,303 @@
+    addParamValue(p,'lon_left',        default_lon_left,        @isnumeric);
+    addParamValue(p,'lon_right',        default_lon_right,        @isnumeric);
+    addParamValue(p,'lat_south',        default_lat_south,        @isnumeric);
+    addParamValue(p,'lat_north',        default_lat_north,        @isnumeric);
+    fprintf('called addParamValues ')
+end
+
+p.parse(nprofiles, varargin{:});
+
+nlevels     = p.Results.nlevels;
+levels      = p.Results.levels;
+T_error_var = p.Results.T_error_var;
+W_error_var = p.Results.W_error_var;
+fill_obs    = p.Results.fill_obs;
+yyyymmdd    = p.Results.YMD;
+nlon        = p.Results.nlon;
+nlat        = p.Results.nlat;
+lon_left        = p.Results.lon_left;
+lon_right        = p.Results.lon_right;
+lat_south        = p.Results.lat_south;
+lat_north        = p.Results.lat_north;
+
+% FIXME ... check that all arrays of length 'level' are consistent.
+
+% If the user specifies nlevels, use the first nlevels of the default.
+% If the user also specifies 'levels', use the first nlevels of what they
+% specify.
+
+nlevels    = min(nlevels,length(levels));
+levels     =       levels(1:nlevels);
+T_error_var = T_error_var(1:nlevels);
+W_error_var = W_error_var(1:nlevels);
+
+levelstrings = sprintf('%.2f ',levels);
+fprintf('vertical levels %s\n',levelstrings)
+
+% Generate obs_sequence input for this problem
+
+% If create_fixed_network_sequence is run, this date information will be overwritten
+
+[year, month, day] = ymd(datetime(yyyymmdd));
+obsdate   = datenum(yyyymmdd);
+gregorian = datenum('1601-01-01');
+dart_date = obsdate-gregorian;
+
+fprintf('Using date/time as %d-%02d-%02d 00:00:00 Z\n',year, month, day)
+fprintf('(days since 1601-01-01) : %f\n',dart_date)
+
+fprintf('Creating %d profiles with %d levels for %d variables = %d observations ...\n', ...
+         nprofiles,nlevels,num_types,nprofiles*nlevels*num_types)
+
+% preallocate space for efficiency
+x(       1:nprofiles) = 0;
+y(       1:nprofiles) = 0;
+z(       1:nprofiles) = 0;
+lon(     1:nprofiles) = 0;   % radians
+lat(     1:nprofiles) = 0;   % radians
+lon_deg( 1:nprofiles) = 0;   % degrees
+lat_deg( 1:nprofiles) = 0;   % degrees
+plot_lon(1:nprofiles) = 0;   % radians, Only num_plot will be filled
+plot_lat(1:nprofiles) = 0;   % radians, Only num_plot will be filled
+
+% Calculate the cartesian and spherical locations.
+% For the geometric and visually minded: see the README.rst.
+inc = pi * (3 - sqrt(5));
+off = 2 / nprofiles;
+for k = 1:nprofiles
+    y(k) = (k-1) * off - 1 + (off / 2);
+    r = sqrt(1 - y(k)*y(k));
+    phi = (k-1) * inc;
+    x(k) = cos(phi) * r;
+    z(k) = sin(phi) * r;
+    lon(k) = atan2(y(k), x(k)) + pi;
+    lat(k) = asin(z(k));
+end
+
+% Now convert latitude and longitude to degrees and restrict to the subdomain.
+
+num_plot = 0;
+for k = 1:nprofiles
+    % Input is in degrees of latitude and longitude; convert from radians
+    lon_deg(k) = rad2deg(lon(k));
```

Comment:
so lon_deg and lat_geg seem to be global, and plot_lon and plot_lat are box bounded.  but when you're writing out the obs, they seem to be using the lon_deg/lat_deg global values?  (line 245).  am i missing something?

Reply:
am still curious if this is a bug or if i'm not reading the code right.  when you used this did you only generate obs inside the specified box?  (or maybe not of interest anymore if we're going with the obs_sequence_tool extraction route.)